src/main/java/hudson/plugins/ircbot/v2/PircListener.java

```diff
@@ -35,189 +40,299 @@
 
-    private final List<PartListener> partListeners = new CopyOnWriteArrayList<PartListener>();
-    
-	
 	volatile boolean explicitDisconnect = false;
-	
-	
+
 	@java.lang.SuppressWarnings("unused")
 	private final PircBotX pircBot;
-    private final String nick;
+	private final String nick;
+	private static final String onlyIndicator = "only";
+	private final String fmtIndicator = " set color ";
 
+	/**
+	 * PircListener constructor
+	 * @param pircBot pircBot
+	 * @param nick nickname of a user
+	 */
 	public PircListener(PircBotX pircBot, String nick) {
-	    this.pircBot = pircBot;
-	    this.nick = nick;
-    }
-
-//	/**
-//	 * {@inheritDoc}
-//	 */
-//    @Override
-//    protected void handleLine(String line) {
-//        LOGGER.fine(line);
-//        super.handleLine(line);
-//    }
-
-    /**
-     * {@inheritDoc} 
-     */
-    @Override
-    public void onMessage(MessageEvent<PircBotX> event) {
-    	for (MessageListener l : this.msgListeners) {
-    		if(l.target.equals(event.getChannel().getName())) {
-    			l.listener.onMessage(new IMMessage(event.getUser().getNick(),
-    			        event.getChannel().getName(), event.getMessage()));
-    		}
-    	}
-    }
-
-    /**
-     * {@inheritDoc} 
-     */
-    @Override
-    public void onPrivateMessage(PrivateMessageEvent<PircBotX> event) {
-        String sender = event.getUser().getNick();
-        String message = event.getMessage();
-    	for (MessageListener l : this.msgListeners) {
-    		if (this.nick.equals(l.target)) {
-    		    if (l.sender == CHAT_ESTABLISHER || sender.equals(l.sender)) {
-    		        l.listener.onMessage(new IMMessage(sender, this.nick, message));
-    		    }
-    		}
-    	}
-    }
-    
-    /**
-     * Someone send me a notice. Possibly NickServ after identifying.
-     */
-    @Override
+		this.pircBot = pircBot;
+		this.nick = nick;
+	}
+
+	/**
+	 * Controls a message sent from PircBotx
+	 * when event has only or format indicator, 
+	 * message is not send it to other event listener, but modifies in the IRC plugin.
+	 * 
+	 * @param event user's is received via PircBotx
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void onMessage(MessageEvent<PircBotX> event) {
+		if (event.getMessage().contains(onlyIndicator)) {
+			onMultiMessage(event);
+		} else if (event.getMessage().contains(fmtIndicator)) {
+			onFormatMessage(event);
+		} else {
+			for (MessageListener l : this.msgListeners) {
+				if (l.target.equals(event.getChannel().getName())) {
+					l.listener.onMessage(new IMMessage(event.getUser()
+							.getNick(), event.getChannel().getName(), event
+							.getMessage()));
+				}
+			}
+		}
+	}
+
+	/**
+	 * Gets usernames from message passed by PircBotx (with "only" flag).
+	 * 
+	 * @param msg
+	 * @param noUser
+	 * @return list of usernames that were passed with the "only" flag
+	 */
+	public static LinkedHashSet<String> getRecipients(String msg, String noUser) {
```

Comment:
Again, please use interfaces instead of implementations.
I think all we need to know here, is that the returned collection is a set, right?